core/common/src/main/java/alluxio/underfs/UnderFileSystem.java

```diff
@@ -277,24 +281,71 @@ public void configureProperties() throws IOException {
   public abstract void close() throws IOException;
 
   /**
+   * Complete the default create operation by renaming temporary path to permanent.
+   *
+   * @param options information to complete create
+   * @throws IOException when rename fails
+   */
+  public void completeCreate(NonAtomicCreateOptions options) throws IOException {
+    String temporaryPath = options.getTemporaryPath();
+    String permanentPath = options.getPermanentPath();
+    if (!rename(temporaryPath, permanentPath)) {
+      if (!delete(temporaryPath, false)) {
+        LOG.error("Failed to delete temporary file {}", temporaryPath);
+      }
+      throw new IOException(
+          ExceptionMessage.FAILED_UFS_RENAME.getMessage(temporaryPath, permanentPath));
+    }
+
+    // Rename does not preserve permissions
```

Comment:
Is this really the case? Previously this code was to ensure any files written by Alluxio services set the correct owner/group as opposed to the Alluxio user/group.